src/qililab/instruments/quantum_machines/quantum_machines_cluster.py

```diff
@@ -597,10 +626,19 @@ def run_compiled_program(self, compiled_program_id: str) -> RunningQmJob:
         Returns:
             RunningQmJob: An object representing the running job. This object provides methods and properties to check the status of the job, retrieve results upon completion, and manage or investigate the job's execution.
         """
-        # CHANGES: qm.queue.add_compiled() -> qm.add_compiled()
-        pending_job = self._qm.queue.add_compiled(compiled_program_id)
-        # CHANGES: job.wait_for_execution() is deprecated and will be removed in the future. Please use job.wait_until("Running") instead.
-        return pending_job.wait_for_execution()  # type: ignore[return-value]
+        # TODO: qm.queue.add_compiled() -> qm.add_compiled()
+        self.pending_job = self._qm.queue.add_compiled(  # pylint: disable=attribute-defined-outside-init
+            compiled_program_id
+        )
+
+        # TODO: job.wait_for_execution() is deprecated and will be removed in the future. Please use job.wait_until("Running") instead.
+        self.job = self.pending_job.wait_for_execution()  # type: ignore[return-value]  # pylint: disable=attribute-defined-outside-init
+        if self.controller == "opx1000" and self._intermediate_frequency:
+            for bus, intermediate_frequency in self._intermediate_frequency.items():
+                self.job.set_intermediate_frequency(element=bus, freq=intermediate_frequency)  # type: ignore[union-attr]
+                self._qm.calibrate_element(bus)
```

Comment:
Shouldn't we reset the `self._intermediate_frequency` after applying the values with the Job API? As it is now, it will apply the settings in every run, right?

Reply:
Thank you for the observation, fixed